### PR TITLE
56 - CRUD for posts, comments, and likes complete

### PR DESCRIPTION
**Pull request by yaaacii**
# [USE_ID] | [CRUD for posts, comments and likes]
> Team member: Cicelia
> Use Case ID:

Fixes #[Issue]

## Expected Behavior
CRUD for posts, comments and likes

## Issue
Needed for posts

### - Usage 
postsGroup := router.Group("posts")	
postsGroup.POST("/", postsController.CreatePost)
        body: handle, postDescription
postsGroup.GET("/", postsController.AllPosts)
postsGroup.GET("/:id", postsController.FindPost)
postsGroup.DELETE("/:id", postsController.DeletePost)
postsGroup.PUT("/:id", postsController.EditPost)
         body: postDescription

	commentsGroup := router.Group("comments")
	commentsGroup.POST("/", commentsController.CreateComment)
              body: handle, commentDescription
	commentsGroup.GET("/", commentsController.AllComments)
	commentsGroup.GET("/:id", commentsController.FindComment)
	commentsGroup.DELETE("/:id", commentsController.DeleteComment)
	commentsGroup.PUT("/:id", commentsController.EditComment)
             body: commentDescription

	likesGroup := router.Group("likes")
	likesGroup.POST("/", likesController.CreateLike)
	likesGroup.GET("/", likesController.AllLikes)
	likesGroup.GET("/:id", likesController.FindLike)
	likesGroup.DELETE("/:id", likesController.DeleteLike)

### - Evidence/Logs
Please include any relevant log snippets or images to support issue.

### CODE REVIEW

**Review by edwardsung4217 — COMMENTED** [review]
Everything looks great, will have you unit test written for these soon?

**Comment by edwardsung4217** [comment]
> 

Never mind, I see that Mikey is writing them. Approved!

**Review by edwardsung4217 — APPROVED** [review]
Looks great, thanks for the work.

File: backend/models/postsModel.go
Comment (Rahgid) [inline backend/models/postsModel.go:1]: Updated the models to use a foreign key. Another example of this is in /models/orgUsersModel.go.

File: backend/models/initialization.go [hunk 1]
```diff
@@ -16,8 +16,9 @@ var tables = []Model{
 	&Users{},
 	&Event{},
 	&Delegations{},
-	// &Friend{},
 	&Posts{},
+	&Comments{},
```
Comment (Rahgid) [inline backend/models/initialization.go:20]: Added these models here to initialize the Comments/Likes tables.

File: backend/models/postsModel.go
Comment (Rahgid) [reply backend/models/postsModel.go:1]: Also made some fields require "NOT NULL" values so we can't create them unless they're specified. Noticed during testing that posts/likes/comments could be added with tons of empty fields. The foreign key requirement also ensures that a valid post exists when a user tries to like/comment on it.

**Review by Rahgid — APPROVED** [review]
Added some changes to the PR, other than this, I think it looks good now. Please just also update the description with the Use Case ID when you get the chance.

Please also add unit tests in the next PR when you get the chance.

**Comment by Rahgid** [comment]
Merging. Again, please just update the Use Case ID and issue tracker info in the PR description.

**Pull request merged**